scripts/snapshot-restore-balance/snapshot-balances.js

```diff
@@ -0,0 +1,86 @@
+const { ApiPromise, WsProvider, Keyring } = require('@polkadot/api');
+const fs = require('fs');
+
+const WS_ENDPOINT = 'wss://testnet-rpc.zkverify.io';
+const DISCARDED_ACCOUNTS = [
+        "5D9txxK9DTvgCznTjJo7q1cxAgmWa83CzHvcz8zhBtLgaLBV",
+        "5ETuZEyLnfVzQCaDM8aQCcsNnz6xjPKvQCtqynCLqwng8QLd",
+        "5D29UEzgStCBTnjKNdkurDNvd3FHePHgTkPEUvjXYvg3brJj",
+        "5DiMVAp8WmFyWAwaTwAr7sU4K3brXcgNCBDbHoBWj3M46PiP",
+        "5EhREncHsntgJaax9YQphk1xN3LxPu2Rzbz4A3g7Ut8cRXWq",
+        "5C84NU2477uHCUF1A8rHb89sP2D2ZsnquPaGa2Htv75FN9gm",
+        "5HdZjrmNAkWQhYQUPNv7YRYnT4vyQswjbNm8eXBvULNQz5wH",
+        "5HjFLKpiCStQgRm6ZM1fT1R5pLKAqQdUG3uh7pvzaQfhdFuB",
+        "5ECktCamcAtBFJirEzvvJmXFxgLMCTAejhqZwLT1Dxn2fwB1",
+        "5EZbvFqx3j7ejqBSPWseif8xL3PwoqMQHdMT8rs9qWoHcdR3"
+    ]
+
+async function takeSnapshot(api, snapshotBalancesFile) {
+    const balances = {};
+    const accounts = await api.query.system.account.keys();
+
+    for (const key of accounts) {
+        const account = key.args[0].toString();
+        if (DISCARDED_ACCOUNTS.includes(account)) continue;
+        const { data: { free: balance } } = await api.query.system.account(account);
+        balances[account] = balance.toString();
+    }
+    const sortedBalances = Object.entries(balances).sort((a, b) => (BigInt(b[1]) > BigInt(a[1]) ? 1 : -1));
+    const sortedBalancesObj = sortedBalances.reduce((acc, [account, balance]) => {
+        acc[account] = balance;
+        return acc;
+    }, {});
+    fs.writeFileSync(snapshotBalancesFile, JSON.stringify(sortedBalancesObj, null, 2));
+
+    console.log('Snapshot saved.');
+}
+
+// Restore balances from the snapshot
+async function restoreBalances(api, keyring, snapshotBalancesFile, custodySeed) {
+    const balances = JSON.parse(fs.readFileSync(snapshotBalancesFile));
+    const custodyAccount = keyring.addFromUri(custodySeed);
+    let nonce = (await api.query.system.account(custodyAccount.address)).nonce.toNumber();
+
+    for (const [account, balance] of Object.entries(balances)) {
+        const transfer = api.tx.balances.transferAllowDeath(account, balance);
+        try {
+            const hash = await transfer.signAndSend(custodyAccount, { nonce });
+            console.log(`Transferred to ${account}, transaction hash: ${hash.toHex()}`);
+            nonce++;
```

Comment:
Isn't the internal nonce of the custody account updated automatically at each signAndSend ?

Reply:
looks like nop, if I remove the optional nonce all the transactions fail with `The transaction has too low priority to replace another transaction already in the pool.`